api-service/database/licenses_test.go

```diff
@@ -0,0 +1,338 @@
+
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar3", "licenseName": "Oracle STD", "purchasedLicense": 0},
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar4", "licenseName": "Oracle STD", "purchasedLicense": 0},
+
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar4", "licenseName": "Real Application Clusters", "purchasedLicense": 1.5},
+
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar3", "licenseName": "WebLogic Server Management Pack Enterprise Edition", "purchasedLicense": 0},
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar4", "licenseName": "WebLogic Server Management Pack Enterprise Edition", "purchasedLicense": 0},
+		}
+
+		assert.JSONEq(t, utils.ToJSON(expectedOut), utils.ToJSON(out))
+	})
+
+	m.T().Run("should_do_pagination", func(t *testing.T) {
+		sortBy := ""
+		sortDesc := false
+		page := 0
+		pageSize := 2
+		location := ""
+		environment := ""
+		olderThan := utils.MAX_TIME
+		out, err := m.db.ListLicenses(sortBy, sortDesc, page, pageSize, location, environment, olderThan)
+		m.Require().NoError(err)
+
+		var expectedOut interface{} = []interface{}{
+			map[string]interface{}{
+				"content": []interface{}{
+					map[string]interface{}{"hostname": "test-db3", "dbName": "foobar3", "licenseName": "Oracle EXE", "purchasedLicense": 0},
+					map[string]interface{}{"hostname": "test-db3", "dbName": "foobar3", "licenseName": "Oracle ENT", "purchasedLicense": 0.5},
+				},
+				"metadata": map[string]interface{}{
+					"empty":         false,
+					"first":         true,
+					"last":          false,
+					"number":        0,
+					"size":          2,
+					"totalElements": 11,
+					"totalPages":    5},
+			},
+		}
+
+		assert.JSONEq(t, utils.ToJSON(expectedOut), utils.ToJSON(out))
+	})
+
+	m.T().Run("should_filter_out_by_location", func(t *testing.T) {
+		sortBy := ""
+		sortDesc := false
+		page := -1
+		pageSize := -1
+		location := "Italy"
+		environment := ""
+		olderThan := utils.MAX_TIME
+		out, err := m.db.ListLicenses(sortBy, sortDesc, page, pageSize, location, environment, olderThan)
+		m.Require().NoError(err)
+
+		var expectedOut interface{} = []interface{}{}
+		assert.JSONEq(t, utils.ToJSON(expectedOut), utils.ToJSON(out))
+	})
+
+	m.T().Run("should_filter_out_by_environment", func(t *testing.T) {
+		sortBy := ""
+		sortDesc := false
+		page := -1
+		pageSize := -1
+		location := ""
+		environment := "TEST"
+		olderThan := utils.MAX_TIME
+		out, err := m.db.ListLicenses(sortBy, sortDesc, page, pageSize, location, environment, olderThan)
+		m.Require().NoError(err)
+
+		var expectedOut interface{} = []interface{}{}
+		assert.JSONEq(t, utils.ToJSON(expectedOut), utils.ToJSON(out))
+	})
+	m.T().Run("should_filter_out_by_older_than", func(t *testing.T) {
+		sortBy := ""
+		sortDesc := false
+		page := -1
+		pageSize := -1
+		location := ""
+		environment := ""
+		olderThan := utils.MIN_TIME
```

Comment:
The parameters take a significant amount of the lines. Why do you not have putted directly into the function parentesis?

Reply:
Removed!